gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceGuiceModule.java

```diff
@@ -200,6 +200,12 @@ public void configure(Binder binder) {
               ServiceConfigKeys.TOPOLOGYSPEC_FACTORY_KEY, ServiceConfigKeys.DEFAULT_TOPOLOGY_SPEC_FACTORY));
     }
 
+    // Initialize flow catalog before DagManager is enabled so templates are loaded when initializing the DagManager
```

Comment:
I'm not too familiar w/ Guice... but are you sure the call to `binder.bind` is guaranteed to synchronously carryout "initializing the flow catalog" (so that finishes before the later call to `binder.bind(DagManager.class)`)?

Reply:
The order does matter here but this is actually the wrong class initialization and won't fix the problem so leaving out of this PR.